src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -511,17 +516,23 @@ def fit_transition(state, dependent_var, independent_var, dependent_var_shape,
     else:
         agg_image_count = float(state)
 
+    x_train = []
+    y_train = []
     # Prepare the data
-    x_train = np_array_float32(independent_var, independent_var_shape)
-    y_train = np_array_int16(dependent_var, dependent_var_shape)
+    for counter, shape in enumerate(independent_var_shape):
+
+        x_train.append(np_array_float32(independent_var[counter], shape))
+    for counter, shape in enumerate(dependent_var_shape):
+
+        y_train.append(np_array_int16(dependent_var[counter], shape))
 
     # Fit segment model on data
     #TODO consider not doing this every time
     fit_params = parse_and_validate_fit_params(fit_params)
     segment_model.fit(x_train, y_train, **fit_params)
 
     # Aggregating number of images, loss and accuracy
-    agg_image_count += len(x_train)
+    agg_image_count += len(x_train[0])
```

Comment:
Checking the counts for every single row should not be necessary. If the counts did not match for some reason, the `fit` would fail before this point with a proper keras error.